benchmarks/luafun/bench.lua

```diff
@@ -0,0 +1,136 @@
+        func.started_aborts = 0
+        func.stitch_roots = 0
+    end
+
+    for i, trace in ipairs(traces) do
+        trace.startloc = bor(lshift(trace.startfunc, 16), trace.startpc)
+
+        local lastlua
+
+        for i = #trace.funcs, 1, -1 do
+            local func = functions[trace.funcs[i]]
+
+            if not func.cfunc then
+                lastlua = func
+                break
+            end
+        end
+
+        trace.lastlua = lastlua
+
+        if lastlua then
+            trace.stoploc = bor(lshift(lastlua.id, 16), trace.stoppc)
+        end
+    end
+
+end
+
+function run_iter(n)
+    for n=1, n do
+        init()
+
+        local aborted = fun.iter(traces):filter(function(event) return event.abort end):totable()
+
+        local completed = fun.iter(traces):filter(function(event) return not event.abort end):totable()
+
+        assert((#aborted + #completed) == #traces)
+
+        roots = fun.iter(completed):
+                    filter(function(event) return event.parentid == 0 end):
+                    totable()
+
+        for i, trace in ipairs(completed) do
+            local startfunc = functions[trace.startfunc]
+            startfunc.starts = startfunc.starts + 1
+
+            -- Function header root trace
+            if trace.parentid == 0 and trace.startpc == 0 and not trace.stitched then
+                assert(trace.exitnum == 0)
+                startfunc.root_trace = trace
+            end
+
+            if trace.parentid == 0 and not trace.stitched then
+                startfunc.sideexits = startfunc.sideexits + 1
+            end
+
+            if trace.stitched then
+                startfunc.stitch_roots = startfunc.stitch_roots + 1
+            end
+
+            local stopfunc = functions[trace.stopfunc]
+            stopfunc.stops = stopfunc.stops + 1
+        end
+
+        for i, trace in ipairs(aborted) do
+            local startfunc = functions[trace.startfunc]
+            startfunc.starts = startfunc.starts + 1
+            startfunc.started_aborts = startfunc.started_aborts + 1
+
+            local stopfunc = functions[trace.stopfunc]
+            stopfunc.aborts = stopfunc.aborts + 1
+
+            local lastlua = trace.lastlua
+            if stopfunc ~= lastlua then
+                lastlua.aborts = lastlua.aborts + 1
+            end
+        end
+
+        local top_aborted = fun.iter(functions):max_by(function(f1, f2) return (f1.aborts > f2.aborts and f1) or f2 end)
+        stats.top_aborted = top_aborted
+        stats.top_aborted_list = fun.iter(aborted):filter(function(trace) return trace.lastlua == top_aborted end):totable()
+        assert(top_aborted.aborts == #stats.top_aborted_list)
+
+        stats.top_stitched = fun.iter(functions):max_by(function(f1, f2) return (f1.stitch_roots > f2.stitch_roots and f1) or f2 end)
+
+        --Build aborted trace stop locations counts
+        local aborted_locations = fun.iter(aborted):reduce(function(locs, trace)

```

Comment:
The indentation of this is really hard to read. At first I thought `end, {})` was a syntax error!

Reply:
The lines can get ridiculously long from call chains and trying to stuff multi-line function on a single line i think they would become unreadable. Even Roberto Ierusalimschy regrets not making a compact style for closures https://www.youtube.com/watch?v=uhBeH3r3FMQ&feature=youtu.be&t=132.

Reply:
I mean that we should do something like:

```lua
local x = function(y)
             blah
          end
```

to at least make the intent clear to readers.